src/main/java/com/aws/greengrass/shadowmanager/model/configuration/ShadowSyncConfiguration.java

```diff
@@ -116,71 +143,121 @@ private static void processOtherThingConfigurations(Map<String, Object> configTo
     private static void processCoreThingConfiguration(Map<String, Object> configTopicsPojo, String thingName,
                                                       Set<ThingShadowSyncConfiguration> syncConfigurationSet) {
         configTopicsPojo.computeIfPresent(CONFIGURATION_CORE_THING_TOPIC, (ignored, coreThingConfigObject) -> {
-            processThingConfiguration(coreThingConfigObject, thingName, syncConfigurationSet);
+            processThingConfiguration(coreThingConfigObject, thingName, syncConfigurationSet,
+                    CONFIGURATION_CORE_THING_TOPIC);
             return coreThingConfigObject;
         });
     }
 
-    private static void processThingConfiguration(Object thingConfigObject, String coreThingName,
+    /**
+     * Processes the thing configuration if presented in a map format.
+     *
+     * @param thingConfigObject    The thing configuration object
+     * @param thingName            The thing name
+     * @param syncConfigurationSet the sync configuration list to add the nucleus thing configuration to.
+     * @throws InvalidRequestParametersException if the named shadow validation fails.
```

Comment:
Which type of exception do you want? I guess its a typo here?

Reply:
I will address this in the next PR.